wgpu-native/src/device.rs

```diff
@@ -253,14 +253,27 @@ impl<B: hal::Backend> PendingResources<B> {
     }
 }
 
+type DeviceToken<'a> = Token<'a, Device<back::Backend>>;
+
 impl PendingResources<back::Backend> {
-    fn triage_referenced(&mut self, trackers: &mut TrackerSet) {
+    fn triage_referenced(&mut self, trackers: &mut TrackerSet, mut token: &mut DeviceToken) {
+        // Before destruction, a resource is expected to have the following strong refs:
+        //  1. in resource itself
+        //  2. in the device tracker
+        //  3. in this list
```

Comment:
I think we need to add a fourth item to this list?

Reply:
actually, no :)
once it has 4 references, it's no longer a resource "before destruction"